.gitlab-ci.yml

```diff
@@ -14,7 +14,7 @@ workflow:
     # In the case of the default branch, we want to build the image so that
     # other branches have a base image to cache from when possible.
     - if: '$CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH'
-    - if: '$CI_EXTERNAL_PULL_REQUEST_IID != null || $CI_EXTERNAL_PULL_REQUEST_IID != ""'
+    - if: '$CI_EXTERNAL_PULL_REQUEST_IID != null && $CI_EXTERNAL_PULL_REQUEST_IID != ""'
```

Comment:
Just curious, do we have evidence that both conditions are really needed here? Seeing a lot of [examples](https://github.com/search?q=ci_external_pull_request_iid+%21%3D&type=code) that just have
```yaml
- if: '$CI_EXTERNAL_PULL_REQUEST_IID 
```
which seems like it might cover all cases?